File: Alexis/Java/Leccion01/MundoPc/src/ar/com/system2023/mundopc/Monitor.java
package ar.com.system2023.mundopc;

public class Monitor {
    private final int idMonitor;
    private String marca;
    private double tamaño;
    private static int contadorMonitores;
    
    private Monitor(){
        this.idMonitor= ++Monitor.contadorMonitores;
    }
    
    public Monitor(String marca, double tamaño){
        this(); //llamado al constructor vacio
        this.marca = marca;
        this.tamaño = tamaño;
    }

    public String getMarca() {
        return this.marca;
    }

    public void setMarca(String marca) {
        this.marca = marca;
    }

    public double getTamaño() {
        return this.tamaño;
    }

    public void setTamaño(double tamaño) {
        this.tamaño = tamaño;
    }
    //ingrasamos manualmente el gutIdMonitor
    public int getIdMonitor(){
        return this.idMonitor;
    }

    @Override
    public String toString() {
        return "Monitor{" + "idMonitor=" + idMonitor + ", marca=" + marca + ", tama\u00f1o=" + tamaño + '}';
    }
    
    
}

File: Alexis/Python/Leccion14/mundo_pc/Monitor.py
class Monitor:
    contador_monitores = 0

    def __init__(self, marca, tamaño):
        Monitor.contador_monitores += 1
        self._id_monitor = Monitor.contador_monitores
        self._marca = marca
        self._tamaño = tamaño

    def __str__(self):
        return f'id: {self._id_monitor}, marca: {self._marca}, tamaño: {self._tamaño}'

if __name__ == '__main__':
    monitor1 = Monitor('hp', '15 pulgadas')
    print(monitor1)
    monitor2 = Monitor('acer', '27 pulgadas')
    print(monitor2)




